src/librustc/mir/interpret/error.rs

```diff
@@ -242,6 +251,9 @@ impl<'tcx> From<InterpError<'tcx>> for InterpErrorInfo<'tcx> {
     }
 }
 
+/// Information about a panic.
+///
+/// FIXME: this is not actually an InterpError, and should probably be moved to another module.
```

Comment:
@oli-obk any proposals where to put this? The `Debug` impl of this is still used by const-eval and const-prop.

Reply:
In that case you can move it to `src/librustc_mir/interpret`. Maybe create an error handling submodule that for now just contains this type.

Reply:
Ah I should have been clearer, the type itself is also used in MIR.^^ So the default place would be `rustc/mir/mod.rs` I think. But should the `Debug` impl also be there?

Reply:
oh... well, if it's part of the MIR, you need to keep the `Debug` impl around for debug printing, as most MIR datastructures can be debug printed.

`rustc/mir/mod.rs` seems fine to me

Reply:
@oli-obk  All right, I moved it. But I also realized the name is kind of bad... this is not about panics in general, it is about assertions in particular. It's just that `AssertMessage` is already a type alias for the specific `PanicInfo<Operand>` that actually gets embedded into MIR.

So what would be a better name? Maybe `AssertInfo`? Though the distinction between `AssertInfo` and `AssertMessage` is entirely not obvious.